### ISSUE TITLE
BUG: np.seterr(invalid='ignore') has to be run twice

### ISSUE THREAD

**Issue by mauroo4** [numpy/numpy#25262]
### Describe the issue:

I am running Python code on Windows 10. Python version: 3.11.5, numpy version: 1.26.0
Problem: running the command
`np.seterr(invalid='ignore')`
Will not have any effect. To get it to work, it has to be run twice.

### Reproduce the code example:

```python
import numpy as np
np.seterr(invalid='ignore')
```

### Runtime information:

1.26.0
3.11.5 | packaged by Anaconda, Inc. | (main, Sep 11 2023, 13:26:23) [MSC v.1916 64 bit (AMD64)]

WARNING: `threadpoolctl` not found in system! Install it by `pip install threadpoolctl`. Once installed, try `np.show_runtime` again for more detailed build information
[{'numpy_version': '1.26.0',
  'python': '3.11.5 | packaged by Anaconda, Inc. | (main, Sep 11 2023, '
            '13:26:23) [MSC v.1916 64 bit (AMD64)]',
  'uname': uname_result(system='Windows', release='10', version='10.0.19045', machine='AMD64')},
 {'simd_extensions': {'baseline': ['SSE', 'SSE2', 'SSE3'],
                      'found': ['SSSE3',
                                'SSE41',
                                'POPCNT',
                                'SSE42',
                                'AVX',
                                'F16C',
                                'FMA3',
                                'AVX2'],
                      'not_found': ['AVX512F',
                                    'AVX512CD',
                                    'AVX512_SKX',
                                    'AVX512_CLX',
                                    'AVX512_CNL',
                                    'AVX512_ICL']}}]
None

**Comment by seberg** [comment]
No, this is not wrong, check the "returns" section of the docs.
